File: LSP.cpp
#include <bits/stdc++.h>
using namespace std;


class Base{
  
  public:
  
  virtual void print() {
      cout<<"This is base class printer"<<endl;
  }
  
  virtual int adder( int a, int b){
      cout<<"Base adder"<<endl;
      return (a+b);
  }
  
  virtual int substractor(int a, int b){
      cout<<"Base substractor"<<endl;
       return (a-b);
  }
    
};

class Derived : public Base{
    
    public:
    
    void print() override{
        cout<<"This is Derived class printer"<<endl;
    }
    
    
    int adder(int a, int b) override{
        cout<<"Derived adder"<<endl;
        return (a+b);
    }
    
    int substractor (int a, int b) override {
        cout<<"Derived substractor"<<endl;
        return (a-b);
    }
    
    int multiplier(int a, int b){
         return (a*b);
    }
    
};

void client(Base* obj){
    
    obj->print();
    cout<<"Addition: "<< obj->adder(10,20)<<endl;
    cout<<"Substraction: "<< obj->substractor(20,10)<<endl;
}

int main() {
    
   Base b1;
   Derived d1;
   
   client(&b1);
   
   return 0;

}

File: LSP.java
class Base {

    public void print() {
        System.out.println("This is base class printer");
    }

    public int adder(int a, int b) {
        System.out.println("Base adder");
        return a + b;
    }

    public int substractor(int a, int b) {
        System.out.println("Base substractor");
        return a - b;
    }
}


class Derived extends Base {

    @Override
    public void print() {
        System.out.println("This is Derived class printer");
    }

    @Override
    public int adder(int a, int b) {
        System.out.println("Derived adder");
        return a + b;
    }

    @Override
    public int substractor(int a, int b) {
        System.out.println("Derived substractor");
        return a - b;
    }

    public int multiplier(int a, int b) {
        return a * b;
    }
}

public class Main {
    
    
    public static void client( Base obj){
        obj.print();
        
        System.out.println("Addition:" + obj.adder(10,20));
        System.out.println("Substraction:" + obj.substractor(200, 100));
        
    }

    public static void main(String[] args) {
        
        Base b1  = new Base();
        Derived d1 = new Derived();
        client(d1);

    }
}